connectors/sources/network_drive.py

```diff
@@ -341,14 +341,18 @@ def create_connection(self):
             port=self.port,
         )
 
-    @cached_property
```

Comment:
Will this have performance impact? Is it used multiple times throughout the sync?

Reply:
The impact is pretty big, seems like, but I feel like it's okay to merge this PR and address it later.